test2code/src/main/kotlin/com/epam/drill/test2code/coverage/CoverageSender.kt

```diff
@@ -115,3 +107,22 @@ class IntervalCoverageSender(
         }
     }
 }
+
+private fun processTotalSizeByte(logger: KLogger): BigInteger {
+    try {
+        val retentionLimit = JvmModuleConfiguration.getCoverageRetentionLimit()
+        val maxMemory = Runtime.getRuntime().maxMemory()
+        if (maxMemory == Long.MAX_VALUE) {
```

Comment:
Hint: try to rewrite to address positive scenario first.
```
if (maxMemoryIsSet) {
// use a fraction of maxMemory
} else {
// use a predefined retention limit
}
```

Reply:
reworked